src/Microsoft.OpenApi/Models/OpenApiRequestBody.cs

```diff
@@ -121,34 +122,41 @@ public IEnumerable<IOpenApiParameter> ConvertToFormDataParameters(IOpenApiWriter
         {
             if (Content == null || !Content.Any())
                 yield break;
-
-            foreach (var property in Content.First().Value.Schema.Properties)
+            var properties = Content.First().Value.Schema?.Properties;
+            if(properties != null)
             {
-                var paramSchema = property.Value.CreateShallowCopy();
-                if ((paramSchema.Type & JsonSchemaType.String) == JsonSchemaType.String
-                    && ("binary".Equals(paramSchema.Format, StringComparison.OrdinalIgnoreCase)
-                    || "base64".Equals(paramSchema.Format, StringComparison.OrdinalIgnoreCase)))
+                foreach (var property in properties)
                 {
-                    var updatedSchema = paramSchema switch {
-                        OpenApiSchema s => s, // we already have a copy
-                        // we have a copy of a reference but don't want to mutate the source schema
-                        // TODO might need recursive resolution of references here
-                        OpenApiSchemaReference r => (OpenApiSchema)r.Target.CreateShallowCopy(),
-                        _ => throw new InvalidOperationException("Unexpected schema type")
+                    var paramSchema = property.Value.CreateShallowCopy();
+                    if ((paramSchema.Type & JsonSchemaType.String) == JsonSchemaType.String
+                        && ("binary".Equals(paramSchema.Format, StringComparison.OrdinalIgnoreCase)
+                        || "base64".Equals(paramSchema.Format, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        var updatedSchema = paramSchema switch
+                        {
+                            OpenApiSchema s => s, // we already have a copy
+                                                  // we have a copy of a reference but don't want to mutate the source schema
+                                                  // TODO might need recursive resolution of references here
+                            OpenApiSchemaReference r => (OpenApiSchema?)r.Target?.CreateShallowCopy(),
```

Comment:
```suggestion
                            OpenApiSchemaReference r when r.Target is not null => (OpenApiSchema)r.Target.CreateShallowCopy(),
                            OpenApiSchemaReference r2 when r.Target is null => throw new InvalidOperationException("Unresolved reference target"),
```

and drop the if you've added below